web-common/src/features/reconcile/files-requests.ts

```diff
@@ -0,0 +1,86 @@
+import { WatchRequestClient } from "@rilldata/web-common/features/reconcile/WatchRequestClient";
+import {
+  getRuntimeServiceGetFileQueryKey,
+  getRuntimeServiceListFilesQueryKey,
+} from "@rilldata/web-common/runtime-client";
+import type {
+  RuntimeServiceWatchFilesParams,
+  V1WatchFilesResponse,
+} from "@rilldata/web-common/runtime-client";
+import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+import type { QueryClient } from "@tanstack/svelte-query";
+
+export function filesRequests(queryClient: QueryClient) {
+  let client: WatchRequestClient<
+    RuntimeServiceWatchFilesParams,
+    V1WatchFilesResponse
+  >;
+
+  return runtime.subscribe((runtime) => {
+    client?.cancel();
+    if (!runtime?.instanceId || !runtime?.host) return;
+
+    client = new WatchRequestClient<
+      RuntimeServiceWatchFilesParams,
+      V1WatchFilesResponse
+    >(
+      `${runtime.host}/v1/instances/${runtime.instanceId}/resources/-/watch`,
+      // TODO: filters
+      {}
+    );
+
+    handleFileResponses(client, queryClient, runtime.instanceId);
```

Comment:
Instead of injecting the `client` here, I'd expect a `handleFileResponse` callback to be applied to each message in the stream along these lines:
```
for await (const res of stream) {
    handleFileResponse(res)
}
```

You could open the stream in this `fileRequests` function rather than opening it in the `handleFileResponses` function.